File: fedorova-t.v-kt-41-22/Models/Teacher.vb
Namespace fedorova_t.v_kt_41_22.Models
    Public Class Teacher
        Public Property Id As Integer
        Public Property FirstName As String = String.Empty
        Public Property LastName As String = String.Empty

        Public Property AcademicDegreeId As Integer?
        Public Property AcademicDegree As AcademicDegree?

        Public Property PositionId As Integer?
        Public Property Position As Position?

        Public Property DepartmentId As Integer?
        Public Property Department As Department?

        Public Property ManagedDepartment As Department? ' Кафедра, которой заведует преподаватель

        Public Property Loads As ICollection(Of Load) = New List(Of Load)()

    End Class
End Namespace

File: fedorova-t.v-kt-41-22/Models/Teacher.cs
using System.ComponentModel.DataAnnotations;
using System.Text.RegularExpressions;

namespace fedorova_t.v_kt_41_22.Models
{
    public class Teacher
    {
        // Id
        public int Id { get; set; }

        //ФИО
        public string FirstName { get; set; }
        public string LastName { get; set; }

        // Ученая степень
        public int DegreeId { get; set; }
        public virtual AcademicDegree AcademicDegree { get; set; }

        // Должность
        public int PositionId { get; set; }
        public virtual Position Position { get; set; }

        // Кафедра
        public int DepartmentId { get; set; }
        public virtual Department Department { get; set; }

        public bool IsValidTeacherFirstName()
        {
            return !string.IsNullOrEmpty(FirstName) &&
                   Regex.IsMatch(FirstName, @"^[a-zA-Zа-яА-ЯёЁ]+$");
        }

        // Проверка фамилии
        public bool IsValidTeacherLastName()
        {
            return !string.IsNullOrEmpty(LastName) &&
                   Regex.IsMatch(LastName, @"^[a-zA-Zа-яА-ЯёЁ]+$");
        }

        // Проверка числовых идентификаторов
        public bool IsValidTeacherDegree()
        {
            return DegreeId > 0;
        }

        public bool IsValidTeacherDepartment()
        {
            return DepartmentId > 0;
        }

        public bool IsValidTeacherPosition()
        {
            return PositionId > 0;
        }

        // Проверка строки, можно ли её преобразовать в число
        public static bool IsValidNumberInput(string input)
        {
            return int.TryParse(input, out int result);
        }

        }
}
